utils/uevent/events.go

```diff
@@ -0,0 +1,58 @@
+package uevent
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/gogoproto/proto"
+	"golang.org/x/exp/maps"
+)
+
+// EmitTypedEvent takes a typed event and emits it
+func EmitTypedEvent(ctx sdk.Context, tev proto.Message) error {
+	event, err := TypedEventToEvent(tev)
+	if err != nil {
+		return err
+	}
+	ctx.EventManager().EmitEvent(event)
+	return nil
```

Comment:
can you document why this is needed at all, because isn't there EmitTypedEvent in the SDK already?

Reply:
done